better-words/message_parser.py

```diff
@@ -0,0 +1,61 @@
+import yaml
+import re
+from os import listdir
+from os.path import splitext
+
+class MessageParser(object):
+  def __init__(self):
+    self.words = {}
+
+    # iterate over all the word categories and load up each yml file
+    for file in listdir('./better-words/words'):
+      category, extension = splitext(file)
+      if extension == '.yml':
+        with open('./better-words/words/' + file, 'r') as stream:
+          self.words[category] = yaml.load(stream)
+
+  # check if the given message contains a word or phrase match
+  # TODO: handle messages that contain multiple matches
+  # TODO: check if the user has opted out
+  def is_match(self, message):
+    for category in self.words:
+      for entry in self.words[category]['entries']:
+        for word in entry['words']:
+          regex = r'\b' + word + r'\b'
+          match = re.search(regex, message.get('text'))
+          if match:
+            return {
+              'message': message,
+              'category': self.words[category],
+              'entry': entry,
+              'word': match.group(0)
+            }
+    return None
+
+  # format a message to be sent to the user
+  def format(self, match):
+    message = match['message']
```

Comment:
We could make a wrapper object for this match response to allow us to access it via dot access. Might do that.